src/app/components/FormElements/TimeSlider.tsx

```diff
@@ -24,7 +24,37 @@ export default class TimeSlider extends React.Component<TimeSliderProps> {
         endTime: moment(minTime).add('minutes', arg[1]).toISOString()
       };
       onTimeChanged(newTimes);
-    }    
+    }
+  }
+
+  componentWillMount() {
+    const {
+      minTime: _minTime,
+      maxTime: _maxTime,
+      startTime: _startTime,
+      endTime: _endTime,
+      onTimeChanged
+    } = this.props;
+    const minTime = moment(_minTime);
```

Comment:
I wonder if we change the names of local variables to be moment instead of Time.  i.e

```ts
const {
      minTime,
      maxTime,
      startTime,
      endTime,
      onTimeChanged
    } = this.props;
    const minMoment = moment(minTime);
    const maxMoment = moment(maxTime);
   // ... etc
```

That way when we see them in code, we'd know that they are moments and not strings or anything like that.

Reply:
I like this idea - why don't I do this as part of my housekeeping branch?